lib/money/ecto/numeric_currency_type.ex

```diff
@@ -0,0 +1,56 @@
+if Code.ensure_compiled?(Ecto.Type) do
+  defmodule Money.Ecto.NumericCurrency.Type do
+    @moduledoc """
+    Provides a type for Ecto to store a currency.
+    The underlying data type is a integer.
+
+    ## Migration Example
+
+        create table(:my_table) do
+          add :currency, :integer
+        end
+
+    ## Schema Example
+
+        schema "my_table" do
+          field :currency, Money.Ecto.NumericCurrency.Type
+        end
+    """
+
+    alias Money.Currency
+
+    if macro_exported?(Ecto.Type, :__using__, 1) do
+      use Ecto.Type
+    else
+      @behaviour Ecto.Type
+    end
+
+    @spec type :: :integer
+    def type, do: :integer
+
+    # @spec cast(Money.t() | String.t() | Integer.t()) :: {:ok, atom()}
+    def cast(val)
+
+    def cast(%Money{currency: currency}) when is_integer(currency), do: {:ok, currency}
```

Comment:
I think this case will never happen since the currency is always an atom.
It is also unclear why it should remain in `{:ok, integer()}`